cmake/app/boilerplate.cmake

```diff
@@ -296,69 +296,70 @@ foreach(root ${BOARD_ROOT})
 
   if(DEFINED SHIELD)
     foreach(s ${SHIELD_AS_LIST})
-      list(REMOVE_ITEM SHIELD ${s})
       list(FIND SHIELD_LIST ${s} _idx)
-      if (NOT _idx EQUAL -1)
-        list(GET shields_refs_list ${_idx} s_path)
-        get_filename_component(s_dir ${s_path} DIRECTORY)
+      if (_idx EQUAL -1)
+        continue()
+      endif()
 
-        # if shield config flag is on, add shield overlay to the shield overlays
-        # list and dts_fixup file to the shield fixup file
-        list(APPEND
-          shield_dts_files
-          ${shield_dir}/${s_path}
+      list(REMOVE_ITEM SHIELD ${s})
+
+      list(GET shields_refs_list ${_idx} s_path)
+      get_filename_component(s_dir ${s_path} DIRECTORY)
+
+      # if shield config flag is on, add shield overlay to the shield overlays
+      # list and dts_fixup file to the shield fixup file
+      list(APPEND
+        shield_dts_files
+        ${shield_dir}/${s_path}
         )
-        list(APPEND
-          shield_dts_fixups
-          ${shield_dir}/${s_dir}/dts_fixup.h
+      list(APPEND
```

Comment:
Is that still possible to have shields fixup?
This is not the case anymore in main tree at least, I'm not sure in case of out-of tree shields
Maybe this could be totally removed (in a dedicated change).

Reply:
I have not been following the DT changes, so I'm not sure.

Reply:
Ok.
So, after a quick look, there is nothing that prevent it, but this mainly depends on drivers (sensors and components available on shields) implementations which should no use them.
Now, this is correctly done in in-tree drivers, but we don't know about potential out-of-tree support.
Anyway, support for fixup should be removed short term with on going changes in device tree support. Support removal in shields could be done at that time when supported is removed globally.